moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planning_manager.cpp

```diff
@@ -147,13 +147,46 @@ bool HybridPlanningManager::initialize()
   hybrid_planning_request_server_ = rclcpp_action::create_server<moveit_msgs::action::HybridPlanner>(
       this->get_node_base_interface(), this->get_node_clock_interface(), this->get_node_logging_interface(),
       this->get_node_waitables_interface(), hybrid_planning_action_name,
+      // Goal callback
       [](const rclcpp_action::GoalUUID& /*unused*/,
          std::shared_ptr<const moveit_msgs::action::HybridPlanner::Goal> /*unused*/) {
         RCLCPP_INFO(LOGGER, "Received goal request");
         return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
       },
-      std::bind(&HybridPlanningManager::hybridPlanningCancelCallback, this, std::placeholders::_1),
-      std::bind(&HybridPlanningManager::hybridPlanningRequestCallback, this, std::placeholders::_1));
+      // Cancel callback
+      [this](std::shared_ptr<rclcpp_action::ServerGoalHandle<moveit_msgs::action::HybridPlanner>> /*unused*/) {
+        // Prevent any new global or local requests from going out
+        stop_hybrid_planning_ = true;
+
+        // Cancel local action
+        local_planner_action_client_->async_cancel_all_goals();
+
+        // Cancel global action
+        global_planner_action_client_->async_cancel_all_goals();
+
+        RCLCPP_INFO(LOGGER, "Received request to cancel goal");
+        return rclcpp_action::CancelResponse::ACCEPT;
```

Comment:
If you want to use the function instead of an inline lambda you can call the function from within the lambda and the compiler will optimize away the lambda.  You can also use `auto` for the type of parameter to the lambda and it should deduce the type if you use it to call another function.

Reply:
that sounds complicated.